agenta-web/src/components/TestSetTable/TestsetTable.tsx

```diff
@@ -115,6 +115,31 @@ const TestsetTable: React.FC<testsetTableProps> = ({mode}) => {
         setSelectedRow(gridRef?.current?.getSelectedNodes())
     }
 
+    const handleExportClick = () => {
+        const csvData = convertToCsvFormat(rowData, columnDefs);
+        const filename = `${testsetName}.csv`;
+        downloadCsv(csvData, filename);
+        // Here, you can add the actual CSV export logic later on.
+    };
+    const convertToCsvFormat = (data, columns) => {
```

Comment:
Can you please make this function more generic and type-safe like this?

```javascript
const convertToCsv = (rows: Record<string, string>[], header: string[]) => {
    const headerRow = header.join(",")
    const remainingRows = rows
        .map((row) => header.map((colName) => row[colName]).join(","))
        .join("\n")
    return `${headerRow}\n${remainingRows}`
}
```

And use it like this
```javascript
const csvData = convertToCsv(
        rowData,
        columnDefs.map((col) => col.field),
)
```